modules/bezug_solarwatt/solarwatt.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+from datetime import datetime, timezone
+import os
+import datetime
+import requests
+import sys
+import traceback
+
+solarwattmethod = int(sys.argv[1])
+speicher1_ip = str(sys.argv[2])
+speicher1_ip2 = str(sys.argv[3])
+
+Debug = int(os.environ.get('debug'))
+myPid = str(os.getpid())
+
+
+def DebugLog(message):
+    local_time = datetime.now(timezone.utc).astimezone()
+    print(local_time.strftime(format="%Y-%m-%d %H:%M:%S") + ": PID: " + myPid + ": " + message)
+
+
+if Debug >= 2:
+    DebugLog('Solarwatt Methode: ' + str(solarwattmethod))
+    DebugLog('Solarwatt IP1: ' + speicher1_ip)
+    DebugLog('Solarwatt IP2: ' + speicher1_ip2)
+
+
+if solarwattmethod == 0:  # Abruf über Energy Manager
+    sresponse = requests.get('http://'+speicher1_ip+'/rest/kiwigrid/wizard/devices', timeout=3).json()
+    if len(str(sresponse)) < 10:
```

Comment:
Der `length` check ist mir etwas suspekt. Der war in der alten Version auch schon drin, aber "damals" war er bevor das json jemals geparst wurde. Ich vermute, dass wenn das der Fall sein sollte, dass dann schon das parsen vorher fehlschlägt. Schöner wäre es zu wissen.

Die Änderung kam ursprünglich mit #1362 rein. Vielleicht kann @Flock82 uns dazu noch erhellen, wann das vorkommen kann.

Wenn nicht würde ich den check weglassen. Code bei dem niemand weiß wofür er da ist steht nur im Weg ;-).

Reply:
Die Solarwatt-Geräte haben die unschöne Eigenschaft immer wieder mal leere Antworten (allerdings mit HTTP 200) zurückzuschicken. Im Shell-Script was das der Fallback, so dass trotzdem ein gültiger Messwert zurückgegeben wird.

Reply:
Danke für die Rückmeldung, das ist sehr hilfreich.

In unserem Fall hier würde dann das `.json()` schon wegen einem Decode-Error eine Exception werfen. Ich denke das wäre OK... Das Script würde aussteigen und 10 Sekunden später gibt es dann neue Werte... Der `len`-check könnte dann weg.